test/integration/traces_test.go

```diff
@@ -58,6 +63,8 @@ func testHTTPTracesCommon(t *testing.T, doTraceID bool) {
 	require.NotEmpty(t, parent.TraceID)
 	if doTraceID {
 		require.Equal(t, traceID, parent.TraceID)
+		children := trace.ChildrenOf(parentID)
+		require.Len(t, children, 1)
```

Comment:
Is there any way to check the correctness of the contents of the children?

Reply:
I think this is confusing how it's worded, I will try to clarify the code.  This is checking if the "parent" (local variable name) has a CHILD_OF Reference struct that matches the "parent-id" (from the traceparent field.)  So it is validating that the parent-id field was applied to the "parent" span.
Hopefully something like this is more clear:
		// Note that parentID is value from traceparent field, not the found parent:
		childOfPID := trace.ChildrenOf(parentID)
		require.Len(t, childOfPID, 1)

Reply:
Ah, I understand now. I think expanding to verify the child traceId as Mario suggested will make it clear. Also, maybe a comment saying that this validates the parent is correct?

Reply:
The child spans are already checked for having a traceID that matches the "parent".
Maybe this is a better comment: ?
// Validate that "parent" is a CHILD_OF the traceparent's "parent-id"